File: fixtures/scale/kotlin/medium/src/calc/parser/OperatorTable.kt
package calc.parser

import calc.errors.EvalException
import calc.util.nearlyZero
import kotlin.math.pow

fun applyOperator(op: Char, left: Double, right: Double): Double = when (op) {
    '+' -> left + right
    '-' -> left - right
    '*' -> left * right
    '/' -> if (nearlyZero(right)) throw EvalException("division by zero") else left / right
    '%' -> left % right
    '^' -> left.pow(right)
    else -> throw EvalException("unknown operator " + op)
}

File: fixtures/scale/dart/medium/lib/parser/operator_table.dart
import '../errors/eval_exception.dart';
import '../util/double_compare.dart';
import 'dart:math' as math;

double applyOperator(String op, double left, double right) {
  switch (op) {
    case '+':
      return left + right;
    case '-':
      return left - right;
    case '*':
      return left * right;
    case '/':
      if (nearlyZero(right)) throw EvalException('division by zero');
      return left / right;
    case '%':
      return left % right;
    case '^':
      return math.pow(left, right).toDouble();
    default:
      throw EvalException('unknown operator $op');
  }
}
